### PR TITLE
new: Add moon toolchain for controlling update checks.

### PR DESCRIPTION
**Pull request by harlequin**
In a cooperate environment most internet resources 
are restriced to access, or company policy
describes that binaries shall have to be stored
in so called artifact products for security
scanning, like Artifacotry or Nexus.

With this feature implementation we are able
to define the locations for current information
json and the on-premise location within the
company.

Changelog: feat

### CODE REVIEW

File: crates/cli/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "moon_cli"
-version = "1.29.0"
+version = "1.29.0-patch0"
```
Comment (milesj) [inline crates/cli/Cargo.toml:3]: This'll need to be changed back before it can be merged.

File: packages/types/src/project-config.ts [hunk 2]
```diff
@@ -66,18 +66,7 @@ export interface ProjectDockerConfig {
...
-	| 'go'
-	| 'javascript'
-	| 'php'
-	| 'python'
-	| 'ruby'
-	| 'rust'
-	| 'typescript'
-	| 'unknown'
-	| string;
+export type LanguageType = 'bash' | 'batch' | 'go' | 'javascript' | 'php' | 'python' | 'ruby' | 'rust' | 'typescript' | 'unknown' | string;
```
Comment (milesj) [inline packages/types/src/project-config.ts:69]: Pretty sure all these changes will fail prettier.

The `just schemas` command will run prettier for you, but you probably need to install node modules with `yarn install`.

File: crates/config/src/workspace/update_config.rs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+use schematic::{validate, Config};
+
+/// Configures how and where updates will be received.
+#[derive(Clone, Config, Debug)]
+pub struct UpdateConfig {
+    /// A secure URL for lookup latest version.
+    #[setting(validate = validate::url_secure, default = "https://launch.moonrepo.app/versions/cli/current")]    
+    pub manifest_url: String,
+    
+    /// A secure URL for downloading moon binary.
```
Comment (milesj) [inline crates/config/src/workspace/update_config.rs:10]: "downloading the moon binary"

File: crates/config/src/workspace/update_config.rs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+use schematic::{validate, Config};
+
+/// Configures how and where updates will be received.
+#[derive(Clone, Config, Debug)]
+pub struct UpdateConfig {
+    /// A secure URL for lookup latest version.
```
Comment (milesj) [inline crates/config/src/workspace/update_config.rs:6]: "A secure URL to lookup the latest version."

**Comment by milesj** [comment]
These changes make sense, the only thing I'm wondering about is if this should actually be in `.moon/toolchain.yml` under a `moon` field.

**Comment by harlequin** [comment]
Think you are right about the `.moon/toolchain.yml', fits better to the roadmap feature

**Manage system packages/dependencies within the toolchain**

Will adapt the pull request.

**Comment by harlequin** [comment]
@milesj I have update the pr according the conversation, can you have a look?

**Comment by milesj** [comment]
Awesome stuff! The other failures can be ignored for now.

**Pull request merged**